api-server/server/boot/donate.js

```diff
@@ -244,23 +250,73 @@ export default function donateBoot(app, done) {
       );
   }
 
-  const pubKey = keys.stripe.public;
+  function addDonation(req, res) {
+    const { user, body } = req;
+
+    if (!user || !body) {
+      return res
+        .status(500)
+        .send({ error: 'User must be signed in for this request.' });
+    }
+    return Promise.resolve(req)
+      .then(
+        user.updateAttributes({
+          isDonating: true
+        })
+      )
+      .then(() => res.status(200).json({ isDonating: true }))
+      .catch(err => {
+        log(err.message);
+        return res.status(500).send({
+          type: 'danger',
+          message: 'Something went wrong.'
+        });
+      });
+  }
+
+  function updatePaypal(req, res) {
+    const { headers, body } = req;
+    return Promise.resolve(req)
+      .then(verifyWebHookType)
+      .then(getAsyncPaypalToken)
+      .then(token => verifyWebHook(headers, body, token, keys.paypal.webhookId))
+      .then(hookBody => updateUser(hookBody, app))
+      .then(() => res.status(200).json({ message: 'received hook' }))
+      .catch(err => {
+        log(err.message);
+        return res.status(200).json({ message: 'received hook' });
```

Comment:
Is the plan to manually check the logs to see what the problem was, rather than to get PayPal to try again?  If so, I think the messages will need more information to allow you to find which donation went wrong.

Also, the reply can go in `finally` if it's always the same.

Reply:
Well I think we should be hooking up something similar to rollbar for this. And the reply is lost by PayPal, I did not say a way that they log it. So we should depend on our logging for this.

Reply:
I created an issue to track your feedback.